poe-tasks/upload-java-connector-tar-file.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+# Uploads the java tar for java connectors.
+# Usage: ./poe-tasks/upload-java-connector-tar-file.sh --name destination-bigquery [--pre-release] [--main-release]
+# You must have set the env var GCS_CREDENTIALS, which contains a JSON-formatted GCP service account key.
```

Comment:
I'm not setting this yet since I have a question: are we okay if I reuse the existing `GCS_CREDENTIALS` for this? or do we want to create a new service account? it looks like this is how we are currently setting this:
```
GCS_CREDENTIALS: ${{ secrets.METADATA_SERVICE_DEV_GCS_CREDENTIALS }}
```